tests/integration/test_dbapi_integration.py

```diff
@@ -158,6 +168,38 @@ def test_array_query_param(trino_connection):
     assert rows[0][0] == "array(integer)"
 
 
+def test_array_timestamp_query_param(trino_connection):
```

Comment:
Could you please add tests also for date and time?

Reply:
And in the tests let's ensure following scenarios are covered:

- negative year (-0001-01-01)
- zero year (0000-01-01)
- gregorian-julian switch (1582-10-04 to 1582-10-14)
- epoch (1970-01-01)
- timestamps that lie in a DST gap (i.e. a timestamp that never occurred in a timezone)
- timestamps that are doubled due to DST

cc: @findepi since timestamps

Reply:
Here are the min max values for python, now I think still it makes sense to support native types as they will handle 99% of normal use cases. We need to decide on how to handle the impossible cases.

```
>>> from datetime import date, time, datetime
>>> date.min
datetime.date(1, 1, 1)
>>> date.max
datetime.date(9999, 12, 31)
>>> time.min
datetime.time(0, 0)
>>> time.max
datetime.time(23, 59, 59, 999999)
>>> datetime.min
datetime.datetime(1, 1, 1, 0, 0)
>>> datetime.max
datetime.datetime(9999, 12, 31, 23, 59, 59, 999999)
```

It seems also supported to use non-existing dates, both in Python as in Trino cfr. https://libguides.ctstatelibrary.org/hg/colonialresearch/calendar

See also See also https://github.com/trinodb/trino/issues/5781

Doubled timestamps are not yet tested.

Reply:
FYI: in snowflake invalid dates just generate exceptions eg snowflake.connector.errors.InterfaceError: 252005: Failed to convert: field DATE '0000-01-01': DATE::-719528, Error: date value out of range

Reply:
We need to provide a flag to control behaviour since this is a regression in client behaviour. Before this change it was possible to use the client to query negative dates or values lying in DST gaps/doubling. Now it's not.